runtime/src/accounts.rs

```diff
@@ -107,26 +115,33 @@ impl Accounts {
                 return Err(TransactionError::AccountLoadedTwice);
             }
 
+            let credit_only_rent_debtors = w_credit_only_rent_debtors.as_mut().unwrap();
+            let mut current_credit_only_pubkeys: HashSet<Pubkey> = HashSet::new();
+
             // There is no way to predict what program will execute without an error
             // If a fee can pay for execution then the program will be scheduled
             let mut accounts: TransactionAccounts = vec![];
             let mut credits: TransactionCredits = vec![];
             let mut rents: TransactionRents = vec![];
-            for key in message
+            for (i, key) in message
                 .account_keys
                 .iter()
                 .filter(|key| !message.program_ids().contains(&key))
+                .enumerate()
             {
                 let (account, rent) = AccountsDB::load(storage, ancestors, accounts_index, key)
-                    .and_then(|(mut account, _)| {
-                        if let (Some(_), rent_collected) = rent_collector.update(&mut account) {
-                            Some((account, rent_collected))
-                        } else {
-                            None
-                        }
-                    })
+                    .and_then(
+                        |(mut account, _)| match rent_collector.update(&mut account) {
+                            (Some(_), rent_due) => Some((account, rent_due)),
```

Comment:
since this always returns Some(something), you can just unwrap(), right?

Reply:
`and_then` will return `None`, if `AccountDB::load` returns None.

Reply:
https://doc.rust-lang.org/std/result/enum.Result.html#method.and_then

Reply:
got it